scripts/acceptance_test.py

```
@@ -0,0 +1,394 @@
+            if r.detail and not r.passed:
+                print(f"         {r.detail}")
+        print()
+        return passed == total
+
+
+def fetch_json(base_url: str, path: str, params: dict | None = None) -> dict:
+    """Fetch JSON from a URL with query parameters."""
+    url = f"{base_url.rstrip('/')}{path}"
+    if params:
+        url = f"{url}?{urllib.parse.urlencode(params)}"
+    req = urllib.request.Request(url, headers={"Accept": "application/json"})
```

Ruff (0.15.9)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)